src/components/ArticleList.jsx

```diff
@@ -8,20 +8,73 @@ function ArticleList() {
   const { topic_slug } = useParams();
   const navigate = useNavigate();
   const [loading, setLoading] = useState(true);
+  const [sort, setSort] = useState();
+  const [ascOrDesc, setascOrDesc] = useState("desc");
+
+  function handleSort(e) {
+    console.log(e.target.innerText);
+    switch (e.target.innerText) {
+      case "Date":
+        setSort("created_at");
+        return;
+      case "Comment count":
+        setSort("comment_count");
+        return;
+      case "Upvotes":
+        setSort("votes");
+        return;
+      case "Newest or Oldest":
+        ascOrDesc === "desc" ? setascOrDesc("asc") : setascOrDesc("desc");
+        return;
+      default:
+        break;
+    }
+  }
 
   useEffect(() => {
-    getArticles(topic_slug).then((articlesRes) => {
+    getArticles(topic_slug, sort, ascOrDesc).then((articlesRes) => {
       setArticles(articlesRes);
       setLoading(false);
     });
-  }, [topic_slug]);
+  }, [topic_slug, sort, ascOrDesc]);
 
   function handleArticleClick(id) {
     navigate(`/article/${id}`);
   }
   if (!loading) {
     return (
       <div className="article--list-box">
+        <section className="sort-by">
+          <h3 className="sort-by-text">Sort by:</h3>
+          <button
+            className="sorting-buttons"
+            onClick={handleSort}
+            disabled={sort === "created_at"}
+          >
+            Date
+          </button>
+          <button
+            className="sorting-buttons"
+            onClick={handleSort}
+            disabled={sort === "comment_count"}
+          >
+            Comment count
+          </button>
+          <button
+            className="sorting-buttons"
+            onClick={handleSort}
+            disabled={sort === "votes"}
+          >
+            Upvotes
+          </button>
+          <button
+            className="sorting-buttons"
+            id="asc-desc"
+            onClick={handleSort}
+          >
+            Newest or Oldest
```

Comment:
This is a bit misleading, as newest/oldest only really applies to date. Maybe change this for ascending/descending. I'd also consider a way of showing users what state it is currently on. Could you have the button text render dynamically to achieve this?